app/src/main/java/com/andela/helpmebuy/adapters/ReceivedRequestAdapter.java

```diff
@@ -0,0 +1,97 @@
+package com.andela.helpmebuy.adapters;
+
+import android.content.Context;
+import android.content.Intent;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import com.andela.helpmebuy.R;
+import com.andela.helpmebuy.activities.PurchaseReqResponse;
+import com.andela.helpmebuy.models.PurchaseItem;
+import com.andela.helpmebuy.models.PurchaseRequest;
+import com.andela.helpmebuy.utilities.DateManager;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by andeladev on 01/06/2016.
+ */
+public class ReceivedRequestAdapter extends RecyclerView.Adapter<ReceivedRequestAdapter.CustomViewHolder> {
+
+    private Context context;
+
+    private List<PurchaseRequest> purchaseRequests;
+
+    public ReceivedRequestAdapter(Context context, List<PurchaseRequest> request) {
+        this.purchaseRequests = request;
+        this.context = context;
+    }
+
+    @Override
+    public CustomViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(parent.getContext())
+                .inflate(R.layout.purchase_request_list, parent, false);
+        return new CustomViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(final CustomViewHolder holder, final int position) {
+        DateManager dateManager = new DateManager();
+        PurchaseRequest request = purchaseRequests.get(position);
+        holder.sender.setText(request.getSendersFullName());
+        String items = getItemsDescription(request);
+        holder.description.setText(items);
+        holder.date.setText(dateManager.formatTime(request.getDateMillis()));
+    }
+
+    public String getItemsDescription(PurchaseRequest request){
+        String itemName = "";
+        ArrayList<PurchaseItem> items = request.getPurchaseList();
+        if(items.size() == 1) {
+            return itemName + items.get(0).getItemName();
+        }
+        else {
```

Comment:
NITPICK: I don't think there is a need to do this and then loop inside. You can just do a multiple check around if `items == 1`, `items > 1 but < 3`, else do the other things.